be/src/exprs/jit/jit_wrapper.cpp

```diff
@@ -0,0 +1,166 @@
+        LOG(ERROR) << "Failed to create LLJIT instance" << error_message;
+        return Status::InternalError("Failed to create LLJIT instance");
+    }
+    _jit = std::move(jit.get());
+
+    // Initialize pass manager for IR optimization.
+    // TODO(Yueyang): check optimization level.
+    // TODO(Yueyang): check if we need to add more passes.
+    _pass_manager_builder.OptLevel = 3;
+    _pass_manager_builder.SLPVectorize = true;
+    _pass_manager_builder.LoopVectorize = true;
+    _pass_manager_builder.VerifyInput = true;
+    _pass_manager_builder.VerifyOutput = true;
+    _pass_manager_builder.populateModulePassManager(_pass_manager);
+
+    _initialized = true;
+    return Status::OK();
+}
+
+void JITWapper::setup_module(llvm::Module* module) const {
+    // Set the data layout of the LLVM module, telling the compiler how to arrange data.
+    module->setDataLayout(*_data_layout);
+    // Set the target triple of the LLVM module to specify the architecture for which the code should be generated.
+    module->setTargetTriple(_target_machine->getTargetTriple().getTriple());
+}
+
+void JITWapper::optimize_module(llvm::Module* module) {
+    // Create a function pass manager.
+    llvm::legacy::FunctionPassManager fpm(module);
+
+    // TODO(Yueyang): check if we need to add more passes.
+    // fpm.add(llvm::createTargetTransformInfoWrapperPass(machine->getTargetIRAnalysis()));
+    // mpm.add(llvm::createTargetTransformInfoWrapperPass(machine->getTargetIRAnalysis()));
+
+    _pass_manager_builder.populateFunctionPassManager(fpm);
+
+    fpm.doInitialization();
+    for (auto& function : *module) {
+        fpm.run(function);
+    }
+    fpm.doFinalization();
+}
+
+void* JITWapper::compile_module(std::unique_ptr<llvm::Module> module, std::unique_ptr<llvm::LLVMContext> context) {
+    // print_module(*module);
+    auto resource_tracker = _jit->getMainJITDylib().createResourceTracker();
+    auto error =
+            _jit->addIRModule(resource_tracker, llvm::orc::ThreadSafeModule(std::move(module), std::move(context)));
+    if (UNLIKELY(error)) {
+        auto* func = lookup_function("evalute");
+        // The function has already been compiled.
+        if (func != nullptr) {
+            LOG(WARNING) << "Failed to add IR module to JIT, module already exist: ";
+            return func;
+        }
+
+        std::string error_message;
+        llvm::handleAllErrors(std::move(error), [&](const llvm::ErrorInfoBase& EIB) { error_message = EIB.message(); });
+        LOG(ERROR) << "Failed to add IR module to JIT: " << error_message;
+        return nullptr;
+    }
+
+    // _resource_tracker_map.emplace("evalute", std::move(resource_tracker));
+    return lookup_function("evalute");
+}
+
+Status JITWapper::remove_module(const std::string& expr_name) {
+    auto it = _resource_tracker_map.find(expr_name);
+    if (it == _resource_tracker_map.end()) {
+        return Status::OK();
+    }
+
+    auto error = it->second->remove();
```

Comment:
should delete it from `_resource_tracker_map`?

Reply:
This is a bug, I will fix it.